hoot-services/src/main/java/hoot/services/models/osm/Way.java

```diff
@@ -313,11 +313,36 @@ public void checkAndFailIfUsedByOtherObjects() throws OSMAPIAlreadyDeletedExcept
             List<Tuple> elementRecords = (List<Tuple>) Element.getElementRecordsWithUserInfo(getMapId(),
                     ElementType.Node, elementIds);
 
+            // It is too slow to do an XPathApi call for every single node to check if it
+            // already exists so its better to just get all nodes and store the ids in list
+            // and check against the list instead.
+            NodeList allNodesList;
+            try {
+                allNodesList = XPathAPI.selectNodeList(doc, "/osm/node");
```

Comment:
Could you just modify the xpath to get the id attribute, instead of doing the for loop below?

Reply:
Yes but the problem is it would be stored in a nodelist which doesnt have a contains function. There is an 'item(int index)' but that isn't very helpful.